src/test/java/org/kiwiproject/dropwizard/util/health/HttpConnectionsHealthCheckTest.java

```diff
@@ -0,0 +1,164 @@
+package org.kiwiproject.dropwizard.util.health;
+
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.assertj.core.api.Assertions.entry;
+import static org.kiwiproject.test.constants.KiwiTestConstants.JSON_HELPER;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.health.HealthCheck;
+import com.fasterxml.jackson.core.type.TypeReference;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.kiwiproject.collect.KiwiMaps;
+import org.kiwiproject.test.dropwizard.mockito.DropwizardMockitoMocks;
+
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+@DisplayName("HttpConnectionsHealthCheck")
+@ExtendWith(SoftAssertionsExtension.class)
+class HttpConnectionsHealthCheckTest {
+
+    private HttpConnectionsHealthCheck healthCheck;
+    private MetricRegistry metrics;
+
+    @BeforeEach
+    void setUp() {
+        var environment = DropwizardMockitoMocks.mockEnvironment();
+        metrics = DropwizardMockitoMocks.mockMetricRegistry(environment);
+        healthCheck = new HttpConnectionsHealthCheck(metrics);
+    }
+
+    @Test
+    void testClientIsNotBeingDumb_ByCheckingForInvalidThreshold(SoftAssertions softly) {
+        softly.assertThatThrownBy(() -> newHealthCheckWithThreshold(0.0)).isExactlyInstanceOf(IllegalArgumentException.class);
+        softly.assertThat(catchThrowable(() -> newHealthCheckWithThreshold(1.0))).isNull();
```

Comment:
This is confusing....why not replace with `assertThatCode(...).doesNotThrowAnyException()` ?

Could put each of these as separate tests in a `@Nested` class